applications/nrf5340_audio/src/bluetooth/bt_rendering_and_capture/volume/bt_vol_ctlr.c

```diff
@@ -70,49 +97,151 @@ static bool vcs_client_peer_exists(struct bt_conn *conn)
 }
 
 /**
- * @brief	Callback handler for the volume state.
+ * @brief Get the index of a given Volume Control Server.
  *
- * @note	This callback handler will be triggered if volume state has changed,
- *		or the playback was muted or unmuted.
+ * @param[in] vcs The given volume control server.
+ * @return int 0 or positive integer on success, -EFAULT if not found.
  */
-static void vcs_state_ctlr_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t volume,
-				      uint8_t mute)
+static int vcs_peer_index_get(struct bt_vcp_vol_ctlr const *const vcs)
 {
-	int ret;
+	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (!vcs_peer_alive(vcs)) {
+			/* Skip */
+			continue;
+		}
 
-	if (err) {
-		LOG_ERR("VCS state callback error: %d", err);
-		return;
+		if (vcs == vcs_client_peer[i].peer) {
+			return i;
+		}
 	}
 
+	return -EFAULT;
+}
+
+/**
+ * @brief Get the common volume of all devices
+ *
+ * @param[out] volume
+ * @return int	0 on success,
+ *		-EFAULT if no common volume was found.
+ *		This can happen if no adjustments have been made or no devices are connected.
+ */
+static int common_vol_get(uint8_t *volume)
```

Comment:
Consider having the same behavior here as you do for `vcs_peer_index_get`, i.e. negative value => error, positive value => return value (volume in this case)